### PR TITLE
Issue #43 Ask for confirmation when delete history

### PR DESCRIPTION
**Pull request by olivierperez**
Show an AlertDialog when user try to delete HTTP calls or Errors to confirm his action.

### CODE REVIEW

File: library/src/main/java/com/readystatesoftware/chuck/internal/ui/error/ErrorListFragment.java [hunk 1]
```diff
@@ -97,6 +98,24 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+        new AlertDialog.Builder(getContext())
+            .setTitle(R.string.chuck_clear)
+            .setMessage(R.string.chuck_clear_error_confirmation)
+            .setPositiveButton(R.string.chuck_clear, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    getContext().getContentResolver().delete(ChuckContentProvider.ERROR_URI, null, null);
+                }
+            })
+            .setNegativeButton(R.string.chuck_cancel, new DialogInterface.OnClickListener() {
```
Comment (cortinico) [inline library/src/main/java/com/readystatesoftware/chuck/internal/ui/error/ErrorListFragment.java]: You can replace here with

```suggestion
            .setNegativeButton(R.string.chuck_cancel, null)
```

And remove all the lines below

File: library/src/main/java/com/readystatesoftware/chuck/internal/ui/transaction/TransactionListFragment.java [hunk 2]
```diff
@@ -126,6 +127,25 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+            .setTitle(R.string.chuck_clear)
+            .setMessage(R.string.chuck_clear_http_confirmation)
+            .setPositiveButton(R.string.chuck_clear, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    getContext().getContentResolver().delete(ChuckContentProvider.TRANSACTION_URI, null, null);
+                    NotificationHelper.clearBuffer();
+                }
+            })
+            .setNegativeButton(R.string.chuck_cancel, new DialogInterface.OnClickListener() {
```
Comment (cortinico) [inline library/src/main/java/com/readystatesoftware/chuck/internal/ui/transaction/TransactionListFragment.java]: As before:

```suggestion
            .setNegativeButton(R.string.chuck_cancel, null)
```

File: library/src/main/java/com/readystatesoftware/chuck/internal/ui/error/ErrorListFragment.java (hunk 1 above)
Comment (olivierperez) [reply library/src/main/java/com/readystatesoftware/chuck/internal/ui/error/ErrorListFragment.java]: Ok

File: library/src/main/java/com/readystatesoftware/chuck/internal/ui/transaction/TransactionListFragment.java (hunk 2 above)
Comment (olivierperez) [reply library/src/main/java/com/readystatesoftware/chuck/internal/ui/transaction/TransactionListFragment.java]: Ok

**Pull request merged**